### PR TITLE
[Cache] Add "Invalidate cache now" button

### LINKED ISSUE #40978: Cache UI: Follow-up work
```[tasklist]
### Follow up work for cache UI
- [x] Measure minimum query duration in seconds (#40970)
- [x] Change default minimum query duration to 1 second (not 60 seconds) (#40970)
- [x] Add 'Invalidate now' button (#40972)
- [x] Move Invalidate now button to the top of the form along with the name of the database (#40972)
- [x] Create DelayedSpinner component (#41103)
- [ ] Add e2e tests. This will be complicated, since our test of the old cache invalidation UI was disabled due to a bug
- [ ] Ensure that form fields can be tabbed into even when behind form buttons div (I have this in a branch but I'm not sure it's worth pursuing)
- [x] Rename TTL to Multiplier
- [x] [not planned] Tweak: remove value from inputs (and just show placeholder) if it matches the default (and ensure form dirty status is correct)
- [ ] Polish: delay all spinners by 300 ms
- [ ] Try Ryan's suggestion to avoid the useVerticallyOverflows hook (https://app.graphite.dev/github/pr/metabase/metabase/39234/Basic-cache-admin-UI?utm_source=gt-slack-notif#comment-PRRC_kwDOAczgH85cTMwl)
- [ ] Explore Ryan's suggestion to use different forms for different strategies (https://app.graphite.dev/github/pr/metabase/metabase/39234/Basic-cache-admin-UI?utm_source=gt-slack-notif#comment-PRRC_kwDOAczgH85cTRiE)
- [ ] https://github.com/metabase/metabase/issues/41119
```

### LINKED ISSUE #41433: Add InvalidateNow button
```[tasklist]
### Tasks
```

### PR DESCRIPTION
**Pull request by rafpaf**
Adds an "Invalidate now" button to the strategy form (in Admin settings > Performance). The button appears when a database is selected, unless the database's strategy is set to "No cached". The button invalidates the cache for the selected database.

Closes #41433

image

### CODE REVIEW

**Comment by rafpaf** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/metabase/metabase/40972?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#40972** <a href="https://app.graphite.dev/github/pr/metabase/metabase/40972?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#40980** <a href="https://app.graphite.dev/github/pr/metabase/metabase/40980?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rafpaf and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by iethree — CHANGES_REQUESTED** [review]
1) the strategy name changes look good, if they were in their own pr, that would be easy to approve
1) functionality also looks and works great
2) I really don't think the invalidation button should be its own form, all the ref passing and portaling are smells that this is a hack and there is a better way. it should just be a button within the form. yes, we'll need a status-updating button that's separate from formik, but i think that is a generally worthwhile endeavor, not just for this bit of UI.

File: enterprise/frontend/src/metabase-enterprise/caching/components/StrategyEditorForDatabases.unit.spec.tsx [hunk 1]
```diff
@@ -27,14 +27,14 @@ describe("StrategyEditorForDatabases", () => {
     const rootStrategyHeading = await screen.findByText("Default policy");
     expect(rootStrategyHeading).toBeInTheDocument();
     expect(
-      await screen.findByLabelText("Edit default policy (currently: Duration)"),
```
[the remark this answers is not in the document]
Comment (rafpaf) [reply enterprise/frontend/src/metabase-enterprise/caching/components/StrategyEditorForDatabases.unit.spec.tsx:30]: yes but there are further changes here not included in that PR

File: frontend/src/metabase/admin/performance/components/StrategyForm.tsx [hunk 2]
```diff
@@ -99,48 +119,144 @@ const StrategyFormBody = ({
...
+}: {
+  targetId: number | null;
+  shouldAllowInvalidation: boolean;
+  invalidateFormContainerRef: React.RefObject<HTMLDivElement>;
+  forceFormUpdate: DispatchWithoutAction;
+}) => {
+  const { dirty } = useFormikContext<Strategy>();
+  const { status } = useFormContext();
+
+  shouldAllowInvalidation &&= targetId !== rootId;
```
[the remark this answers is not in the document]
Comment (rafpaf) [reply frontend/src/metabase/admin/performance/components/StrategyForm.tsx:234]: good call

**Review by iethree — APPROVED** [review]
looks good!

one little thing: should the "done" state of the invalidation button be clickable?

**Pull request merged**